sql/plan/ddl.go

```diff
@@ -614,44 +604,50 @@ func (d *DropTable) WithTriggers(triggers []string) sql.Node {
 }
 
 // TableNames returns the names of the tables to drop.
-func (d *DropTable) TableNames() []string {
-	return d.names
+func (d *DropTable) TableNames() ([]string, error) {
+	tblNames := make([]string, len(d.Tables))
+	for i, t := range d.Tables {
+		// either *ResolvedTable OR *UnresolvedTable here
+		if uTable, ok := t.(*UnresolvedTable); ok {
+			tblNames[i] = uTable.Name()
+		} else if rTable, ok := t.(*ResolvedTable); ok {
+			tblNames[i] = rTable.Name()
+		} else {
+			return []string{}, sql.ErrInvalidType.New(t)
+		}
+	}
+	return tblNames, nil
+}
+
+// IfExists returns ifExists variable.
+func (d *DropTable) IfExists() bool {
+	return d.ifExists
 }
 
 // RowIter implements the Node interface.
 func (d *DropTable) RowIter(ctx *sql.Context, row sql.Row) (sql.RowIter, error) {
-	droppable, ok := d.db.(sql.TableDropper)
-	if !ok {
-		return nil, sql.ErrDropTableNotSupported.New(d.db.Name())
-	}
-
 	var err error
-	for _, tableName := range d.names {
-		tbl, ok, err := d.db.GetTableInsensitive(ctx, tableName)
+	var curdb sql.Database
 
-		if err != nil {
-			return nil, err
-		}
+	for _, table := range d.Tables {
+		tbl, _ := table.(*ResolvedTable)
+		curdb = tbl.Database
 
-		if !ok {
-			if d.ifExists {
-				continue
-			}
+		droppable, _ := tbl.Database.(sql.TableDropper)
```

Comment:
No need to use the `_` here

If the assertion isn't possible it will just panic